tokio/src/util/wake_list.rs

```diff
@@ -37,12 +37,34 @@ impl WakeList {
     }
 
     pub(crate) fn wake_all(&mut self) {
+        struct DropGuard {
+            wakers: *mut Waker,
+            start: usize,
+            end: usize,
+        }
+
+        impl Drop for DropGuard {
+            fn drop(&mut self) {
+                let slice = ptr::slice_from_raw_parts_mut(
+                    unsafe { self.wakers.add(self.start) },
```

Comment:
This unsafe block needs a safety comment. It should explain that the resulting pointer is in-bounds (or one after the end) of the `self.wakers` array.